src/charm_state.py

```diff
@@ -345,6 +347,18 @@ def from_charm(  # pylint: disable=too-many-arguments
             # ignoring because mypy fails with:
             # "has incompatible type "**dict[str, str]"; expected ...""
             valid_synapse_config = SynapseConfig(**dict(charm.config.items()))  # type: ignore
+            # remove workers from instance_map
+            if instance_map_config and valid_synapse_config.workers_ignore_list:
+                logger.debug(
+                    "Removing %s from instance_map", valid_synapse_config.workers_ignore_list
+                )
+                for worker in valid_synapse_config.workers_ignore_list.split(","):
```

Comment:
May I suggest doing a `.trim()` for each worker as well? since the users may write "workerA, workerB, workerC, ..."